src/models/link.es6.js

```diff
@@ -5,6 +5,34 @@ class Link extends Base {
   constructor(props) {
     delete props.selftext_html;
 
+    if (props.promoted) {
```

Comment:
An ad model would be preferable to modifying the link model (it could inherit from `Link` and call `super` in the constructor to get all of the other logic, too.)